server/consumer.go

```diff
@@ -701,6 +705,39 @@ func (o *Consumer) processNak(sseq, dseq uint64) {
 	}
 }
 
+// Process a TERM
+func (o *Consumer) processTerm(sseq, dseq, dcount uint64) {
+	// Treat like an ack to suppress redelivery.
+	o.processAckMsg(sseq, dseq, dcount, false)
+
+	o.mu.Lock()
+	defer o.mu.Unlock()
+
+	// Deliver an advisory
+	e := JSConsumerDeliveryTerminatedAdvisory{
```

Comment:
why not use the existing exceeded delivery? This basically is a short cut to hitting exceeded? If we wanted to indicate early termination it could be 1 extra field there rather than a whole new advisory

Reply:
Fair question as that was what we had discussed but it feels like a very different thing. Maybe we can prefix so easy to capture both?

If you feel super strong I can change over to use the other but does not feel like a delivery exceeded event. Very different IMO.

Reply:
yeah just from the initial discussion, and I think its cognitive overhead to think of even more ways and more reporting, monitoring etc - they are close enough related that its a easy thing to explain?

Donno, I can go eitherway so whatever you like :)

Reply:
I think separate..